Auto update failing for xkeyboard-config

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the [xkeyboard-config](https://github.com/termux/termux-packages/blob/ece9f646ff0bfbc6d04944e0da0b159910810dc6/x11-packages/xkeyboard-config) package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="bash">
INFO: Updating xkeyboard-config [Current version: 2.41]
INFO: package being updated to 2.42.

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  888k  100  888k    0     0  4367k      0 --:--:-- --:--:-- --:--:-- 4357k

You are about to commit these changes:

--------------------
diff --git a/x11-packages/xkeyboard-config/build.sh b/x11-packages/xkeyboard-config/build.sh
index d0243331b..6b5d568b4 100644
--- a/x11-packages/xkeyboard-config/build.sh
+++ b/x11-packages/xkeyboard-config/build.sh
@@ -4,9 +4,9 @@ TERMUX_PKG_DESCRIPTION="X keyboard configuration files"
 TERMUX_PKG_LICENSE="custom"
 TERMUX_PKG_LICENSE_FILE="COPYING"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="2.41"
+TERMUX_PKG_VERSION="2.42"
 TERMUX_PKG_SRCURL=https://xorg.freedesktop.org/archive/individual/data/xkeyboard-config/xkeyboard-config-${TERMUX_PKG_VERSION}.tar.xz
-TERMUX_PKG_SHA256=f02cd6b957295e0d50236a3db15825256c92f67ef1f73bf1c77a4b179edf728f
+TERMUX_PKG_SHA256=a6b06ebfc1f01fc505f2f05f265f95f67cc8873a54dd247e3c2d754b8f7e0807
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_PLATFORM_INDEPENDENT=true
 
--------------------

bump(x11/xkeyboard-config): 2.42


Not committing to Git!

INFO: Trying to build package.
INFO: Package xkeyboard-config exists in x11 repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building xkeyboard-config for arch aarch64...
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13133  100 13133    0     0   189k      0 --:--:-- --:--:-- --:--:--  191k
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   1326      0 --:--:-- --:--:-- --:--:--  1328
gpg: Signature made Sat 08 Jun 2024 06:16:39 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/main/binary-aarch64/Packages
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
  4 1582k    4 71864    0     0  20293      0  0:01:19  0:00:03  0:01:16 20289
  7 1582k    7  125k    0     0  30625      0  0:00:52  0:00:04  0:00:48 30624
 12 1582k   12  189k    0     0  34664      0  0:00:46  0:00:05  0:00:41 35991
 17 1582k   17  269k    0     0  42847      0  0:00:37  0:00:06  0:00:31 52720
 23 1582k   23  365k    0     0  50953      0  0:00:31  0:00:07  0:00:24 72831
 30 1582k   30  477k    0     0  58978      0  0:00:27  0:00:08  0:00:19 87774
 42 1582k   42  669k    0     0  74479      0  0:00:21  0:00:09  0:00:12  108k
 62 1582k   62  989k    0     0  99233      0  0:00:16  0:00:10  0:00:06  173k
 95 1582k   95 1517k    0     0   134k      0  0:00:11  0:00:11 --:--:--  259k
100 1582k  100 1582k    0     0   140k      0  0:00:11  0:00:11 --:--:--  309k
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13319  100 13319    0     0  23925      0 --:--:-- --:--:-- --:--:-- 23955
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   1361      0 --:--:-- --:--:-- --:--:--  1361
100   833  100   833    0     0   1361      0 --:--:-- --:--:-- --:--:--  1361
gpg: Signature made Thu 06 Jun 2024 12:23:28 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13119  100 13119    0     0  22563      0 --:--:-- --:--:-- --:--:-- 22580
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   1373      0 --:--:-- --:--:-- --:--:--  1372
gpg: Signature made Sat 08 Jun 2024 01:42:18 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://xorg.freedesktop.org/archive/individual/data/xkeyboard-config/xkeyboard-config-2.42.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  888k  100  888k    0     0  4690k      0 --:--:-- --:--:-- --:--:-- 4702k
Downloading https://github.com/mesonbuild/meson/releases/download/1.4.0/meson-1.4.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 2172k  100 2172k    0     0  3592k      0 --:--:-- --:--:-- --:--:-- 3592k
[termux_setup_meson]: Applying meson-libintl.patch
[termux_setup_meson]: Applying meson-python.patch
[termux_setup_meson]: Applying meson-soversion.patch
DEPRECATION: "pkgconfig" entry is deprecated and should be replaced by "pkg-config"
The Meson build system
Version: 1.4.0
Source dir: /home/builder/.termux-build/xkeyboard-config/src
Build dir: /home/builder/.termux-build/xkeyboard-config/build
Build type: cross build
Project name: xkeyboard-config
Project version: 2.42
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: aarch64
Host machine cpu: aarch64
Target machine cpu family: aarch64
Target machine cpu: aarch64
Found pkg-config: YES (/home/builder/.termux-build/_cache/android-r26b-api-24-v4/bin/pkg-config) 0.29.2
Program generate-options-symbols.py found: YES (/home/builder/.termux-build/xkeyboard-config/src/rules/generate-options-symbols.py)
Program compat/map-variants.py found: YES (/home/builder/.termux-build/xkeyboard-config/src/rules/compat/map-variants.py)
Program merge.py found: YES (/home/builder/.termux-build/xkeyboard-config/src/rules/merge.py)
Program xml2lst.pl found: YES (/home/builder/.termux-build/xkeyboard-config/src/rules/xml2lst.pl)
Program xsltproc found: YES (/usr/bin/xsltproc)
Configuring man.xsl using configuration
Program tests/verify-group-names.sh found: YES (/home/builder/.termux-build/xkeyboard-config/src/tests/verify-group-names.sh)
Program python3 (pytest) found: NO
Program pytest-3 pytest found: NO
Program msgfmt found: YES (/usr/bin/msgfmt)
Program msginit found: YES (/usr/bin/msginit)
Program msgmerge found: YES (/usr/bin/msgmerge)
Program xgettext found: YES (/usr/bin/xgettext)
Build targets in project: 66
NOTICE: Future-deprecated features used:
 * 0.64.0: {'copy arg in configure_file'}

xkeyboard-config 2.42

  User defined options
    Cross files        : /home/builder/.termux-build/xkeyboard-config/tmp/meson-crossfile-aarch64.txt
    buildtype          : minsize
    libdir             : lib
    prefix             : /data/data/com.termux/files/usr
    strip              : true
    compat-rules       : true
    xkb-base           : /data/data/com.termux/files/usr/share/X11/xkb
    xorg-rules-symlinks: true

Found ninja-1.12.1 at /home/builder/.termux-build/_cache/ninja-1.12.1/ninja
[1/63] Generating rules/0012-ml_s.part with a custom command
[2/63] Generating rules/0010-mlv_s.part with a custom command
[3/63] Generating rules/0043-option_compat.part with a custom command (wrapped by meson to capture output)
FAILED: rules/0043-option_compat.part 
/usr/bin/python3 /home/builder/.termux-build/_cache/meson-1.4.0/meson.py --internal exe --capture rules/0043-option_compat.part -- /home/builder/.termux-build/xkeyboard-config/src/rules/generate-options-symbols.py --rules-section=compatibility --xkb-config-root /home/builder/.termux-build/xkeyboard-config/src ../src/rules/base.xml ../src/rules/base.extras.xml
--- stderr ---
Traceback (most recent call last):
  File "/home/builder/.termux-build/xkeyboard-config/src/rules/generate-options-symbols.py", line 10, in <module>
    from enum import StrEnum, unique
ImportError: cannot import name 'StrEnum' from 'enum' (/usr/lib/python3.10/enum.py)

[4/63] Generating rules/0042-option_symbols.part with a custom command (wrapped by meson to capture output)
FAILED: rules/0042-option_symbols.part 
/usr/bin/python3 /home/builder/.termux-build/_cache/meson-1.4.0/meson.py --internal exe --capture rules/0042-option_symbols.part -- /home/builder/.termux-build/xkeyboard-config/src/rules/generate-options-symbols.py --rules-section=symbols --xkb-config-root /home/builder/.termux-build/xkeyboard-config/src ../src/rules/base.xml ../src/rules/base.extras.xml
--- stderr ---
Traceback (most recent call last):
  File "/home/builder/.termux-build/xkeyboard-config/src/rules/generate-options-symbols.py", line 10, in <module>
    from enum import StrEnum, unique
ImportError: cannot import name 'StrEnum' from 'enum' (/usr/lib/python3.10/enum.py)

[5/63] Generating rules/0044-option_types.part with a custom command (wrapped by meson to capture output)
FAILED: rules/0044-option_types.part 
/usr/bin/python3 /home/builder/.termux-build/_cache/meson-1.4.0/meson.py --internal exe --capture rules/0044-option_types.part -- /home/builder/.termux-build/xkeyboard-config/src/rules/generate-options-symbols.py --rules-section=types --xkb-config-root /home/builder/.termux-build/xkeyboard-config/src ../src/rules/base.xml ../src/rules/base.extras.xml
--- stderr ---
Traceback (most recent call last):
  File "/home/builder/.termux-build/xkeyboard-config/src/rules/generate-options-symbols.py", line 10, in <module>
    from enum import StrEnum, unique
ImportError: cannot import name 'StrEnum' from 'enum' (/usr/lib/python3.10/enum.py)

[6/63] Generating rules/0014-ml1_s.part with a custom command
ninja: build stopped: subcommand failed.
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2024-06-08 18:18:59 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/9430585568">9430585568</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/x11-packages/xkeyboard-config/build.sh
+++ b/x11-packages/xkeyboard-config/build.sh
@@ -1,14 +1,16 @@
-TERMUX_PKG_HOMEPAGE=https://www.freedesktop.org/wiki/Software/XKeyboardConfig/
+TERMUX_PKG_HOMEPAGE=https://gitlab.freedesktop.org/xkeyboard-config/xkeyboard-config
 TERMUX_PKG_DESCRIPTION="X keyboard configuration files"
 # Licenses: HPND, MIT
 TERMUX_PKG_LICENSE="custom"
 TERMUX_PKG_LICENSE_FILE="COPYING"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="2.41"
-TERMUX_PKG_SRCURL=https://xorg.freedesktop.org/archive/individual/data/xkeyboard-config/xkeyboard-config-${TERMUX_PKG_VERSION}.tar.xz
-TERMUX_PKG_SHA256=f02cd6b957295e0d50236a3db15825256c92f67ef1f73bf1c77a4b179edf728f
+TERMUX_PKG_VERSION="2.42"
+TERMUX_PKG_SRCURL=https://gitlab.freedesktop.org/xkeyboard-config/xkeyboard-config/-/archive/xkeyboard-config-${TERMUX_PKG_VERSION}/xkeyboard-config-xkeyboard-config-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=932ff62af3245ad248a3b659a4b1435354acdae7273bbbc61fd183fddb2e4748
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_PLATFORM_INDEPENDENT=true
+TERMUX_PKG_BUILD_DEPENDS="python"
+TERMUX_PKG_PYTHON_COMMON_DEPS="StrEnum"
 
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
 -Dxkb-base=${TERMUX_PREFIX}/share/X11/xkb

